src/enrichment/babamul.rs

```diff
@@ -5,40 +5,185 @@ use crate::enrichment::lsst::EnrichedLsstAlert;
 use crate::enrichment::ztf::EnrichedZtfAlert;
 use crate::enrichment::EnrichmentWorkerError;
 use apache_avro::{Schema, Writer};
-use std::collections::HashMap;
+use schemars::schema_for;
+use serde_json::Value as JsonValue;
+use std::collections::{BTreeMap, HashMap};
 use tracing::error;
 
-// Avro schemas for enriched LSST and ZTF alerts sent to Babamul
-// TODO: These are just placeholders for now and need to be defined properly
-const ENRICHED_LSST_ALERT_SCHEMA: &str = r#"
-{
-    "type": "record",
-    "name": "EnrichedLsstAlert",
-    "fields": [
-        {"name": "candid", "type": "long"},
-        {"name": "objectId", "type": "string"},
-        {"name": "candidate", "type": "string"},
-        {"name": "prv_candidates", "type": "string"},
-        {"name": "fp_hists", "type": "string"},
-        {"name": "properties", "type": "string"}
-    ]
+// Generate a small Avro schema from a Rust type inspected via `schemars`.
+// We only include the fields used by Babamul (the fields that are put into
+// the Avro record). Complex nested Rust types (objects/arrays) are mapped
+// to Avro `string` since we serialize those nested values as JSON strings
+// when producing the Avro payload.
+// Convert a schemars JSON Schema node to an Avro JSON type.
+fn json_schema_node_to_avro(
```

Comment:
Did you look at https://crates.io/crates/apache-avro-derive (struct -> avro schema), or https://crates.io/crates/avrotize (json schema -> avro schema)? Might be easier than re-implementing it yourself from scratch

Reply:
Switching over to that now. Thanks for the tip!